src/main/java/com/uber/jenkins/phabricator/PhabricatorNotifier.java

```diff
@@ -312,48 +311,58 @@ private CoverageProvider getCoverageProvider(Run<?, ?> build, FilePath workspace
         }
 
         Logger logger = new Logger(listener.getLogger());
-        InstanceProvider<CoverageProvider> provider = new InstanceProvider<CoverageProvider>(
-                Jenkins.getInstance(),
-                COBERTURA_PLUGIN_NAME,
-                COBERTURA_CLASS_NAME,
-                logger
-        );
-        CoverageProvider coverage = provider.getInstance();
+        List<CoverageProvider> coverageProviders = new ArrayList<CoverageProvider>();
 
-        if (coverage == null) {
-            return null;
+        CoverageProvider coberturaCoverage = makeProvider(COBERTURA_PLUGIN_NAME, COBERTURA_CLASS_NAME, logger);
+        if (coberturaCoverage != null) {
+            coverageProviders.add(coberturaCoverage);
         }
 
-        coverage.setBuild(build);
-        coverage.setWorkspace(workspace);
-        coverage.setIncludeFileNames(includeFileNames);
-        coverage.setCoverageReportPattern(coverageReportPattern);
-        if (coverage.hasCoverage()) {
-            return coverage;
-        } else {
-            logger.info(UBERALLS_TAG, "No cobertura results found");
-            return null;
+        CoverageProvider jacocoCoverage = makeProvider(JACOCO_PLUGIN_NAME, JACOCO_CLASS_NAME, logger);
+        if (jacocoCoverage != null) {
+            coverageProviders.add(jacocoCoverage);
+        }
+
+        for (Iterator<CoverageProvider> i = coverageProviders.iterator(); i.hasNext(); ) {
+            CoverageProvider coverageProvider = i.next();
+            coverageProvider.setBuild(build);
+            coverageProvider.setWorkspace(workspace);
+            coverageProvider.setIncludeFileNames(includeFileNames);
+            coverageProvider.setCoverageReportPattern(coverageReportPattern);
+
+            if (!coverageProvider.hasCoverage()) {
+                i.remove();
+            }
+        }
+        if (!coverageProviders.isEmpty()) {
```

Comment:
Can we add a `logger.info` statement here about which coverage we picked if its a non empty list?

It will help with debugging issues